### ISSUE TITLE
model.evaluate() is giving the same result for different numpy arrays

### ISSUE THREAD

**Issue by isa-tr** [serengil/deepface#566]
Hello, 

I am currently evaluating DeepFace Emotion model as follows:

import tensorflow as tf

DeepFace_Model.compile(optimizer='adam', loss=tf.keras.losses.CategoricalCrossentropy() ,metrics=['accuracy'])
evaluation = DeepFace_Model.evaluate(norm_x_test_batch,y_test_enc)

norm_x_test_batch is a batch of images which were processed as deepface does and y_test_enc are the true labels for those images.

When I run DeepFace_Model.evaluate the result is always the same even if I change the batch of images. 

![image](https://user-images.githubusercontent.com/82721764/192643278-d5e088dc-021c-479e-ad27-f163f0d3e418.png)

![image](https://user-images.githubusercontent.com/82721764/192643371-47dfe079-e944-4053-8c23-c4aa122e6a00.png)

Those are different experiments.

**Comment by serengil** [comment]
whwn you compile the model, then it loses the pre-trained weights.

**Comment by isa-tr** [comment]
Hello, thank you for your answer.

Although, when I run model.evaluate() without compiling it, it yields an error that says:

![image](https://user-images.githubusercontent.com/82721764/193293322-16b3f394-04c2-4f2f-a8de-cc0a0b77c6a1.png)

What I'm trying to evaluate is the performance of DeepFace Emotion model with a customized test dataset of mine.

**Comment by serengil** [comment]
I recommend you to write your own evaluator. Built model will be able to return predictions with .predict command for your data set.

**Comment by isa-tr** [comment]
Okay thank you :)
